README.md

```diff
@@ -27,3 +27,9 @@ about the current recursive call.
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. You do not need to prove that the invariant is correct. Add
 your answer to this markdown file.
+
+I made sure that certain indexes would be defined with the proper values if needed. The amount of these given values changes depending on the input into the function. Also, some of these variiables will remain undefined until after the final recursive call (this would be the invariant). This is true as there is always some amount of undefined variables throughout the code, until the array is slowly filled with actually defined values. But making sure that some are defined at the beginning allows for the rest of the calculations to be done properly with a good starting point. ChatGPT did help with some of this reasoning, as well as the ideas of slowly filling in the array.
+
+The Invariant is as follows: At least one element within the array will be undefined until after the final recursive call. This can be used to check if the algorithm has been executed properly, by constantly checking to see if there are any undefined values within the array. If there are no undefined values when this is expected, then you can know that the algorithm was executed properly.
```

Comment:
But that doesn't tell you anything about the elements that are not undefined. How do you know that you'll have the correct result?